AppCenterDataStorage/AppCenterDataStorageTests/MSDBDocumentStoreTests.m

```diff
@@ -202,6 +200,109 @@ - (void)testReadUserDocumentFromLocalDatabaseNotFound {
   XCTAssertEqualObjects(documentWrapper.documentId, documentId);
 }
 
+- (void)testReadFromLocalDatabaseDeletesDocumentIfNewTTLIsZero {
+
+  // If
+  NSString *documentId = @"12829";
+  NSString *eTag = @"398";
+  NSString *jsonString = @"{ \"document\": {\"key\": \"value\"}}";
+  NSString *pendingOperation = kMSPendingOperationReplace;
+  [self addJsonStringToTable:jsonString
+                        eTag:eTag
+                   partition:self.appToken.partition
+                  documentId:documentId
+            pendingOperation:pendingOperation
+              expirationTime:(long)[[NSDate dateWithTimeIntervalSinceNow:1000000] timeIntervalSince1970]];
+  MSReadOptions *readOptions = [[MSReadOptions alloc] initWithDeviceTimeToLive:MSDataStoreTimeToLiveNoCache];
+
+  // When
+
+  // Read once and update the TTL to expire immediately, then read again to ensure the document is expired.
+  [self.sut readWithToken:self.appToken documentId:documentId documentType:[MSMockDocument class] readOptions:readOptions];
+
+  // Then
+  OCMVerify([self.sut deleteWithToken:self.appToken documentId:documentId]);
+}
+
+- (void)testReadFromLocalDatabaseCorrectlyUpdatesTTLToInfinity {
+
+  // If
+  NSString *documentId = @"12829";
```

Comment:
Can we extract this into a method for adding some generic json to the table?  There's quite a few methods that do exactly this or something very similar

Reply:
I don't think we need to extract this, it's just setup code and I would prefer not to obscure it with more levels of indirection